presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetBinaryColumnReader.java

```diff
@@ -54,7 +54,7 @@ protected void readValue(BlockBuilder blockBuilder, Type type)
             }
             type.writeSlice(blockBuilder, value);
         }
-        else {
+        else if (valueIsNull()) {
             blockBuilder.appendNull();
         }
```

Comment:
For non required fields there are three options:
1) Value is defined
2) Value is null
3) Value does not exist, because one of it's optional parent fields is null.